tests/Functional/Schema/PostgreSQLSchemaManagerTest.php

```diff
@@ -494,6 +494,88 @@ public function testAlterTableAutoIncrementIntToBigInt(string $from, string $to,
         self::assertTrue($tableFinal->getColumn('id')->getAutoincrement());
     }
 
+    public function testListTableDetailsWithTwoSameTablesOnDifferentSchemas(): void
+    {
+        $sql = 'CREATE TABLE migrations(a text);';
+        $this->connection->executeStatement($sql);
+        $sql = 'CREATE TABLE "001_test".migrations(a text);';
+        $this->connection->executeStatement($sql);
+        $this->connection->executeStatement('SET search_path TO "001_test",public');
+
+        $databaseTable = $this->schemaManager->listTableDetails('migrations');
+
+        self::assertNotNull($databaseTable);
```

Comment:
`$databaseTable` is never null: https://github.com/doctrine/dbal/blob/b4f49c050462878fb39423762054fe65276c1f99/src/Schema/AbstractSchemaManager.php#L319-L331